src/Reflection/ReflectionFunctionAbstract.php

```diff
@@ -619,4 +619,17 @@ public function getReturnStatementsAst()
 
         return $visitor->getReturnNodes();
     }
+
+    private function findTypeFromAst($namespace, $locatedSource, $type)
+    {
+        $objectType = (new FindTypeFromAst())->__invoke(
+            $this->reflector->getContextFactory()->createForNamespace(
```

Comment:
> in which case the ContextFactory (or Context perhaps?) 

I think we will always need a `Context` and `ContextFactory`, we could inject the context on creation of the `Reflection[Thing]` but 

--

btw, this change would also mean injecting the context factory to all the methods, parameters etc. of the reflection, but I guess that was always the case, and it is better that the Reflector would no longer contain the `ContextFactory` - which leads me to wonder if, it is ever refactored in the way I suggested, if the `Reflector` should not also be responsible for instantiating the methods, parameters, etc, but I digress :)